### DISCUSSION TITLE
Permissions request on mobile

### DISCUSSION THREAD

**Discussion by c0st1nus** [tauri-apps/tauri#12039] [Q&A]
Hi everyone,  
Just for fun, I made a small project where I want to capture data from the camera, microphone, and geoposition. On desktop, requesting device access through the website was straightforward, and I used the following JavaScript code (Next.js):  
```javascript
useEffect(() => {
    async function getCameraStream() {
        try {
            const stream = await navigator.mediaDevices.getUserMedia({ video: true });
            videoRef.current.srcObject = stream;
            setPermissionGranted(true);
        } catch (err) {
            console.error("Error accessing camera: ", err);
            setError("Permission denied or no camera available.");
        }
    }
    getCameraStream();
}, []);
```
This worked fine on desktop. However, I found an official "Barcode Scanner" plugin for mobile, but I didn’t fully understand how the permission request for the camera is implemented there.  
Could anyone guide me on how to request camera access permission on mobile devices when using Next.js? Are there best practices or specific examples I should follow?  
I have already read the issue #4434 . But... I got nothing from it. By that question it would excellent if someone could explain how to request permissions overall.
Thanks in advance!

**Comment by FabianLars** [comment]
moved to https://github.com/tauri-apps/tauri/issues/12041
